ports/raspberrypi/common-hal/rp2pio/StateMachine.c

```diff
@@ -674,13 +674,26 @@ static bool _transfer(rp2pio_statemachine_obj_t *self,
     size_t len = MAX(out_len, in_len);
     bool tx = data_out != NULL;
     bool rx = data_in != NULL;
-    if (len >= dma_min_size_threshold) {
+    bool use_dma = len >= dma_min_size_threshold || swap_out || swap_in;
```

Comment:
Is there any risk that having DMAs for sizes less then the threshold that require swaps will cause either a speed issue for the small DMAs or potential exhaustion of DMAs? It may not be any concern just wanted to bring it up.

Rest looks great.

Reply:
I don't really know. :)  There are 12 DMAs; audio might claim 2 in the background, and this code now claims up to 2. so it looks like there should be plenty left over.

At first I wanted to say this was no change for existing code, but it IS a change. Before, if you had a LARGE buffer but couldn't allocate DMA, the code would fall back to copying in a CPU loop. Now, it will return an error.